include/crocoddyl/core/solvers/box-qp.hpp

```diff
@@ -29,10 +31,10 @@ struct BoxQPSolution {
   std::vector<size_t> clamped_idx;
 };
 
-// Based on Yuval Tassa's BoxQP
```

Comment:
Why did you remove the doc from here? I found it useful to have this doc to understand what was being developed!

Reply:
@wxmerkt implemented using this code. Then I re-wrote it from scratch without checking or reviewing that code. I had to removed earlier, but I forgot it.

Reply:
I could add citation to Tassa paper (or better the Projected Newton QP paper that Tassa is citing)

Reply:
@wxmerkt what is your opinion about it?

I removed it because this is not a copy-translation of Tassa's code, and it seems to me that we should remove this comment.

Reply:
I compared the implementations a few weeks back and there is little difference: polynomial vs linear line-search (makes no difference in convergence) and LpNorm vs L2Norm for convergence check (no difference in convergence). I think there is a minor change in the Armijo criterion for accepting a step. And of course there's the change from inverse() to LLT. 

We should definitely include a reference (also in general add more comments to code - it's often quite an effort to read Crocoddyl code and follow what's going on) - if you worked off the Projected Newton QP cited in Tassa, that should be included. Otherwise we should leave the Tassa reference.

Reply:
I have included a preliminary doxygen documentation of this class. See my PR.